### ISSUE TITLE
JSON-API calls do not include linked resources when no other parameters are passed

### ISSUE THREAD

**Issue by Stevo-S** [flexirest/flexirest#132]
### Expected behaviour

When I use the `json_api` proxy to make a request for the resource including linked resources,
the actual HTTP request does not contain an `include` parameter if the method call to make 
the request is parameterless.

### Desired behaviour

It should add an `include` parameter for the requested linked resource to the HTTP request
regardless of whether arguments are passed to the method called or not.

### How to reproduce

1. Clone the Flexirest repo (`git clone https://github.com/flexirest/flexirest.git`)
2. Change to the flexirest folder (`cd flexirest`)
3. Install any dependencies (`bundle install`)
4. Run an interactive console (`rake c`)
5. Paste the following:

```ruby

class Article < Flexirest::Base
  proxy :json_api
  
  base_url 'https://enj762y66xib8.x.pipedream.net'
  
  get 'all', '/articles'
end
```

Call the `all` method without passing any arguments but make sure to include `:images`

```ruby
Article.includes(:images).all
=> #<Article success: true, Status: 200>
```

Go to https://requestbin.com/r/enj762y66xib8/1iwbEgYfkfx59wQrn01nO44irz5 to inspect the resulting HTTP request.

You'll see that Request Bin reports the request as

##### GET `/articles`

instead of the expected

##### GET `/articles?include=images`

However, if you pass an argument to the `all` method for example:

```ruby
Article.includes(:images).all(param1: 'some_value')
=> #<Article success: true, Status: 200>
```

and check Request Bin, the request will have the appropriate `include` parameter.

##### GET `/articles?include=images&param1=some_value`

Therefore, the linked resources are not included in the HTTP request if no other parameters are also included.

**Comment by andyjeffries** [comment]
Unfortunately I never use JSON-API so don't know how this area works, maybe @mikevoets or @francois-ferrandis that have submitted PRs on that functionality before could help?

**Comment by francois-ferrandis** [comment]
@andyjeffries I'm on it! Almost ready, I will be submitting a PR in the next few days.

**Comment by andyjeffries** [comment]
Thank you so much @francois-ferrandis ! 💙

**Comment by andyjeffries** [comment]
I've just released this as v1.10.2
